deeplearning4j/dl4j-integration-tests/src/test/java/org/deeplearning4j/integration/testcases/samediff/SameDiffMLPTestCases.java

```diff
@@ -152,4 +172,275 @@ public int getOverfitNumIterations() {
+                File f = new ClassPathResource("dl4j-integration-tests/data/moon_data_eval.csv").getFile();
+                RecordReader rr = new CSVRecordReader();
+                rr.initialize(new FileSplit(f));
+                DataSetIterator iter = new RecordReaderDataSetIterator(rr, 32, 0, 2);
+                return new MultiDataSetIteratorAdapter(iter);
+            }
+
+            @Override
+            public IEvaluation[] doEvaluationSameDiff(SameDiff sd, MultiDataSetIterator iter, IEvaluation[] evaluations) {
+                sd.evaluate(iter, "out", 0, evaluations);
+                return evaluations;
+            }
+
+            @Override
+            public MultiDataSet getOverfittingData() throws Exception {
+
+                File f = new ClassPathResource("dl4j-integration-tests/data/moon_data_eval.csv").getFile();
+                RecordReader rr = new CSVRecordReader();
+                rr.initialize(new FileSplit(f));
+                return new RecordReaderDataSetIterator(rr, 1, 0, 2).next().toMultiDataSet();
+            }
+
+            @Override
+            public int getOverfitNumIterations() {
+                return 200;
+            }
+        };
+    }
+
+    public static TestCase getLenetMnist() {
+        return new TestCase() {
+            {
+                testName = "LenetMnistSD";
+                testType = TestType.RANDOM_INIT;
+                testPredictions = true;
+                testTrainingCurves = true;
+                testGradients = true;
+                testParamsPostTraining = true;
+                testEvaluation = true;
+                testOverfitting = false;
+            }
+
+            @Override
+            public ModelType modelType() {
+                return ModelType.SAMEDIFF;
+            }
+
+            public Object getConfiguration() throws Exception {
+                int nChannels = 1; // Number of input channels
+                int outputNum = 10; // The number of possible outcomes
+                int seed = 123;
+
+
+                //Define the network structure:
+                SameDiff sd = SameDiff.create();
+                SDVariable in = sd.placeHolder("in", DataType.FLOAT, nChannels, 784);
+                SDVariable label = sd.placeHolder("label", DataType.FLOAT, nChannels, outputNum);
+
+                SDVariable w0 = sd.var("w0", Nd4j.rand(DataType.FLOAT, 784, 20));
+
+                SDVariable w1 = sd.var("w1", Nd4j.rand(DataType.FLOAT, 20, 50));
+                SDVariable b1 = sd.var("b2", Nd4j.rand(DataType.FLOAT, 20));
+
+
+                SDVariable w4 = sd.var("w2", Nd4j.rand(DataType.FLOAT, 50, 500));
+                SDVariable b4 = sd.var("b2", Nd4j.rand(DataType.FLOAT, 50));
+
+                SDVariable w5 = sd.var("w3", Nd4j.rand(DataType.FLOAT, 500, outputNum));
+                SDVariable b5 = sd.var("b3", Nd4j.rand(DataType.FLOAT, 500));
+
+
+                SDVariable layer0 = sd.cnn.conv1d("layer0", in, w0, new Conv1DConfig());
```

Comment:
We should be using conv2d here, not conv1d.

But before we can do that, we need a reshape to go to 4d [NCHW] format, from "flattened" 2d format `[minibatch, C*H*W]`:
SDVariable in4d = in.reshape(-1, 1, 28, 28);